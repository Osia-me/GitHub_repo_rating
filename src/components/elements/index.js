export { default as Star } from './Star';
export { default as Logo } from './Logo';
