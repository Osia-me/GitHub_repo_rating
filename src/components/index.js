export * from './elements';
export * from './templates';
export * from './lists';
