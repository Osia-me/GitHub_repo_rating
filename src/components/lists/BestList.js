import React, { useContext } from 'react';
import Card from '../card/Card';
import { Context } from '../../Store.js';
import './BestList.css';

function BestList() {
  const [state, setState] = useContext(Context);
  const arrayOfItems = state.items;
  const favList = state.fav;

  if(!state.view){
    return (
      <div className="list">
        {arrayOfItems ? arrayOfItems.map((repo) =>
          <Card
            key={repo.id}
            id={repo.id}
            name={repo.name}
            description={repo.description}
            numberOfStars={repo.stargazers_count}
            link={repo.html_url} />
          ) : null}
      </div>
    );
  } else {
    return (
      <div className="list">
        {favList ? favList.map((repo) =>
          <Card
            key={`${repo.id}`}
            id={repo.id}
            name={repo.name}
            description={repo.description}
            numberOfStars={repo.stargazers_count}
            link={repo.html_url} />
          ) : null}
      </div>
    );
  }
}

export default BestList;
