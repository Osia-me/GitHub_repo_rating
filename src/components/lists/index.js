export { default as BestList } from './BestList';
