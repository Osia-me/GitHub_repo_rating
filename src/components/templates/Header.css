.header {
  width:auto;
  height:60px;
  margin-bottom: 14px;
  padding-left: 14px;
  padding-right: 14px;
  background-color: #2c333e;
  opacity:0.9;
  position:sticky;
  top:0;
  z-index: 1;
  -webkit-backdrop-filter: blur(8px);
  -moz-backdrop-filter: blur(8px);
  backdrop-filter: blur(8px);
  border-radius: 0px 0px 15px 15px;
}

.header-nav {
  width:100%;
  height:100%;
  display:flex;
  justify-content: space-between;
  align-items: center;
  color: white;
}

.header-name {
  margin: 1em;
  font-size: 1.5rem;
  font-weight: bold;
  vertical-align:8px;
}

.header-right {
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
}

.header-language {
  border-color: white;
  padding: 5px;
  margin: 1em;
  cursor:pointer;
}
