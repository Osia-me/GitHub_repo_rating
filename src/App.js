import { Header } from './components/templates';
import { BestList } from './components/lists';
import Store from './Store';

import './App.css';

function App() {

  return (
    <Store className="app">
      <Header />
      <BestList />
    </Store>
  );
}

export default App;
