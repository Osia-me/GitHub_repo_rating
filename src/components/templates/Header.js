import React, { useState, useEffect, useContext } from 'react';
import Star from '../elements/Star/Star';
import Logo from '../elements/Logo/Logo';
import Language from '../elements/Language/Language';
import { Context } from '../../Store.js';
import './Header.css'

function Header() {
  const [state, setState] = useContext(Context);
  const [clickStar, setClickStar] = useState(0);

  const StarClickHandler = () => {
    clickStar === 0 ? setClickStar(1) : setClickStar(0);
  }

  useEffect(() => {
    if(clickStar){
      setState({...state, view: 'starred'});
    } else{
      setState({...state, view: null});
    }
  },[clickStar, setState]);

  return (
      <header className='header'>
        <nav className='header-nav grow'>
          <Logo />
          <div className='header-right'>
            <Star width={35} height={35} starclick={StarClickHandler} color = {clickStar ? 'color' : null}/>
            <Language />
          </div>

        </nav>
      </header>
  );
}

export default Header;
