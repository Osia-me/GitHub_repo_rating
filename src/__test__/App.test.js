import { render, screen } from '@testing-library/react';
import { shallow } from 'enzyme';
import App from '../App.js';
import { Header } from '../components/templates';
import { BestList } from '../components/lists';

describe('Parent Component', () => {
  it('renders Child component', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.containsMatchingElement(<Header />)).toEqual(true);
    expect(wrapper.containsMatchingElement(<BestList />)).toEqual(true);
  });
});
