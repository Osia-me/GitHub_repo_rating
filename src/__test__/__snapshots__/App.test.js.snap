// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parent Component renders Child component 1`] = `ShallowWrapper {}`;
